docs/examples/tutorial/string/slicing_c_string.pyx

```diff
@@ -0,0 +1,15 @@
+from cpython.mem cimport PyMem_Free
+from c_func cimport get_a_c_string
+
+
+def main():
+    cdef char* c_string = NULL
+    cdef Py_ssize_t length = 0
+
+    # get pointer and length from a C function
+    get_a_c_string(&c_string, &length)
+
+    try:
+        py_bytes_string = c_string[:length]  # Performs a copy of the data
+    finally:
+        PyMem_Free(c_string)

```

Comment:
Ah, no, please use plain `malloc()` and `free()` here. The intention is to show how to interact with something C-ish (e.g. an external C library), not with something that understands the CPython C-API.

Reply:
My bad, I though it would be a good example. I'll change it.